src/common/jsonb/src/functions.rs

```diff
@@ -110,6 +110,156 @@ pub fn build_object<'a>(
+                    buf.extend_from_slice(&encoded.to_be_bytes());
+                    if val_offset > prev_val_offset {
+                        buf.extend_from_slice(&value[prev_val_offset..val_offset]);
+                    }
+                }
+            }
+            Some(buf)
+        }
+        _ => None,
+    }
+}
+
+// get the inner value by name of `JSONB` object
+pub fn get_by_name(value: &[u8], name: &str) -> Option<Vec<u8>> {
+    let mut buf: Vec<u8> = Vec::new();
+
+    let header = read_u32(value, 0).unwrap();
+    match header & CONTAINER_HEADER_TYPE_MASK {
+        OBJECT_CONTAINER_TAG => {
+            let length = (header & CONTAINER_HEADER_LEN_MASK) as usize;
+            let mut jentry_offset = 4;
+            let mut val_offset = 8 * length + 4;
+
+            let mut key_jentries: VecDeque<JEntry> = VecDeque::with_capacity(length);
+            for _ in 0..length {
+                let encoded = read_u32(value, jentry_offset).unwrap();
+                let key_jentry = JEntry::decode_jentry(encoded);
+
+                jentry_offset += 4;
+                val_offset += key_jentry.length as usize;
+                key_jentries.push_back(key_jentry);
+            }
+
+            let mut key_offset = 8 * length + 4;
+            while let Some(key_jentry) = key_jentries.pop_front() {
+                // read value jentry
+                let val_encoded = read_u32(value, jentry_offset).unwrap();
+                let val_jentry = JEntry::decode_jentry(val_encoded);
+
+                let old_key_offset = key_offset;
+                key_offset += key_jentry.length as usize;
+                let key_str = std::str::from_utf8(&value[old_key_offset..key_offset]).unwrap();
+                // compare key and name, if equal return the value
+                if name.eq(key_str) {
+                    let prev_val_offset = val_offset;
+                    val_offset += val_jentry.length as usize;
+
+                    match val_jentry.type_code {
+                        CONTAINER_TAG => {
+                            buf.extend_from_slice(&value[prev_val_offset..val_offset]);
+                        }
+                        _ => {
+                            let scalar_header = SCALAR_CONTAINER_TAG;
+                            buf.extend_from_slice(&scalar_header.to_be_bytes());
+                            buf.extend_from_slice(&val_encoded.to_be_bytes());
+                            if val_offset > prev_val_offset {
+                                buf.extend_from_slice(&value[prev_val_offset..val_offset]);
+                            }
+                        }
+                    }
+                    return Some(buf);
+                }
+                jentry_offset += 4;
+                val_offset += val_jentry.length as usize;
+            }
+            None
+        }
+        _ => None,
+    }
+}
+
+// get the keys of a `JSONB` object
```

Comment:
We can make them doc comments

```suggestion
/// Get the keys of a `JSONB` object
```